File: src/swift-camera/PpgSignal.swift
import Foundation

// Swift equivalent of PpgSignal.cs
class PpgSignalPack {
    var phoneModel: String?
    var phoneOSVersion: String?
    var infoSupportedHardwareLevel: String?
    var ppgSignals: [PpgSignal]?
    var calibration: Calibration?
    
    init() {}
}

class PpgSignal {
    var timestamp: UInt64
    var signals: [Double]
    var qualityWarning: Bool
    
    init(timestamp: UInt64, signals: [Double], qualityWarning: Bool) {
        self.timestamp = timestamp
        self.signals = signals
        self.qualityWarning = qualityWarning
    }
}

class Calibration {
    var dia: Int
    var sys: Int
    var type: String?
    
    init(dia: Int, sys: Int, type: String? = nil) {
        self.dia = dia
        self.sys = sys
        self.type = type
    }
}

File: src/original-c#-camera/PpgSignal.cs
using System;
using System.Collections.Generic;

namespace RivaDigitalApp
{
    public class PpgSignalPack
    {
        public string PhoneModel { get; set; }

        public string PhoneOSVersion { get; set; }

        public string InfoSupportedHardwareLevel { get; set; }

        public IEnumerable<PpgSignal> PpgSignals { get; set; }

        public Calibration Calibration { get; set; }
    }

    public class PpgSignal
    {
        public ulong Timestamp { get; set; }

        public double[] Signals { get; set; }

        public bool QualityWarning { get; set; }
    }

    public class Calibration
    {
        public int Dia { get; set; }

        public int Sys { get; set; }

        public string Type { get; set; }
    }

   
}
